src/Process/PointCloudProcessing.js

```diff
@@ -163,18 +163,49 @@ function markForDeletion(elt) {
 }
 
 export default {
-    preUpdate(context, layer) {
-        // TODO: use changeSource
-        layer.counters = {
-            pointCount: 0,
-            displayedCount: 0,
-        };
-
+    preUpdate(context, layer, changeSources) {
         // Bail-out if not ready
         if (!layer.root) {
             return [];
         }
 
+        if (changeSources.has(undefined) || changeSources.size == 0) {
+            return [layer.root];
+        }
+
+        // lookup lowest common ancestor of changeSources
+        let commonAncestorName;
+        for (const source of changeSources.values()) {
+            if (source.isCamera) {
+                // if the change is caused by a camera move, no need to bother
+                // to find common ancestor: we need to update the whole tree:
+                // some invisible tiles may now be visible
+                return [layer.root];
+            }
+            if (source.obj === undefined) {
```

Comment:
I don't remember where this `obj` came from?
it's an Object3D?

Reply:
See: https://github.com/iTowns/itowns/blob/master/src/Process/PointCloudProcessing.js#L228